File: ArraysAndHashing/contains-duplicate.js
/**
 * https://leetcode.com/problems/contains-duplicate/
 *
 * Time O(n) | Space O(n)
 *
 * hash set to get unique values in array,
 * to check for duplicates easily
 *
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  const seen = new Set();

  for (const num of nums) {
    if (seen.has(num)) return true;
    seen.add(num);
  }
  return false;
};

File: ArraysAndHashing/contains-duplicate/contains-duplicate.cpp
/**
 * https://leetcode.com/problems/contains-duplicate/
 *
 * Time O(n) | Space O(n)
 *
 * hash set to get unique values in array,
 * to check for duplicates easily
 */
#include <vector>
#include <unordered_set>
using namespace std;

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        unordered_set<int> seen;
        for (int num : nums)
        {
            if (seen.count(num))
            {
                return true;
            }
            seen.insert(num);
        }
        return false;
    }
};
;